Cosmetic improvements to InlineDisplay*Builder coordinate mapping

fantasai opened the pull request:
#### ce9e73cd2b64b18416c8e25d001915eb022c62fd
<pre>
Cosmetic improvements to InlineDisplay*Builder coordinate mapping
<a href="https://bugs.webkit.org/show_bug.cgi?id=285275">https://bugs.webkit.org/show_bug.cgi?id=285275</a>
<a href="https://rdar.apple.com/problem/142227210">rdar://problem/142227210</a>

Reviewed by Alan Baradlay.

- Adopt line-relative terminology to shorten variable and function names.
- Use &quot;logical left&quot; and &quot;logical right&quot; consistent with their usage in RenderBlock etc.
- Pass around writingMode objects directly; and clarify whose writingMode it is.
- Reorganize InlineDisplayLineBuilder::build to organize its calculations better;
  and in particular to build a logicalRect that can in the future be passed to a
  standard mapping function

* Source/WebCore/layout/formattingContexts/inline/display/InlineDisplayContentBuilder.cpp:
(WebCore::Layout::marginLineLeft):
(WebCore::Layout::marginLineRight):
(WebCore::Layout::borderLineLeft):
(WebCore::Layout::borderLineRight):
(WebCore::Layout::paddingLineLeft):
(WebCore::Layout::paddingLineRight):
(WebCore::Layout::InlineDisplayContentBuilder::adjustVisualGeometryForDisplayBox):
(WebCore::Layout::InlineDisplayContentBuilder::processBidiContent):
(WebCore::Layout::InlineDisplayContentBuilder::setLogicalLeft const):
(WebCore::Layout::InlineDisplayContentBuilder::setLogicalRight const):
(WebCore::Layout::marginLeftInInlineDirection): Deleted.
(WebCore::Layout::marginRightInInlineDirection): Deleted.
(WebCore::Layout::borderLeftInInlineDirection): Deleted.
(WebCore::Layout::borderRightInInlineDirection): Deleted.
(WebCore::Layout::paddingLeftInInlineDirection): Deleted.
(WebCore::Layout::paddingRightInInlineDirection): Deleted.
(WebCore::Layout::InlineDisplayContentBuilder::setLeftForWritingMode const): Deleted.
(WebCore::Layout::InlineDisplayContentBuilder::setRightForWritingMode const): Deleted.
* Source/WebCore/layout/formattingContexts/inline/display/InlineDisplayContentBuilder.h:
* Source/WebCore/layout/formattingContexts/inline/display/InlineDisplayLineBuilder.cpp:
(WebCore::Layout::InlineDisplayLineBuilder::build const):

Canonical link: <a href="https://commits.webkit.org/288380@main">https://commits.webkit.org/288380@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3d2602e7eb70ef0568b2a5cf4fe0201d04cc63fa

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/82955 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/131/builds/2601 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/55/builds/37249 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/88069 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/34006 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/85048 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/130/builds/2675 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/10519 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/88069 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/34006 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/86011 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/130/builds/2675 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/55/builds/37249 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/88069 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/130/builds/2675 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/55/builds/37249 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/33040 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/130/builds/2675 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/55/builds/37249 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/89435 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/128/builds/10242 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/7430 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/89435 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/10470 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/55/builds/37249 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/89435 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/55/builds/37249 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/1621 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12828 "Built successfully and passed tests") | [❌ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/10195 "Failed to checkout and rebase branch from PR 38448") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/15707 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/10067 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/13532 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/11835 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/layout/formattingContexts/inline/display/InlineDisplayLineBuilder.cpp:
```diff
@@ -124,30 +124,30 @@ InlineDisplayLineBuilder::EnclosingLineGeometry InlineDisplayLineBuilder::collec
-    auto lineBoxLogicalRect = lineBox.logicalRect();
-    auto lineBoxVisualLeft = isLeftToRightDirection
+
+    InlineRect lineBoxLogicalRect = lineBox.logicalRect();
```
`    auto lineBoxLogicalRect = lineBox.logicalRect();`

fantasai replied:
I wanted to make it more explicit that we're taking a copy and not a reference here.

fantasai committed:
Cosmetic improvements to InlineDisplay*Builder coordinate mapping
https://bugs.webkit.org/show_bug.cgi?id=285275
rdar://problem/142227210

Reviewed by Alan Baradlay.

- Adopt line-relative terminology to shorten variable and function names.
- Use "logical left" and "logical right" consistent with their usage in RenderBlock etc.
- Pass around writingMode objects directly; and clarify whose writingMode it is.
- Reorganize InlineDisplayLineBuilder::build to organize its calculations better;
  and in particular to build a logicalRect that can in the future be passed to a
  standard mapping function

* Source/WebCore/layout/formattingContexts/inline/display/InlineDisplayContentBuilder.cpp:
(WebCore::Layout::marginLineLeft):
(WebCore::Layout::marginLineRight):
(WebCore::Layout::borderLineLeft):
(WebCore::Layout::borderLineRight):
(WebCore::Layout::paddingLineLeft):
(WebCore::Layout::paddingLineRight):
(WebCore::Layout::InlineDisplayContentBuilder::adjustVisualGeometryForDisplayBox):
(WebCore::Layout::InlineDisplayContentBuilder::processBidiContent):
(WebCore::Layout::InlineDisplayContentBuilder::setLogicalLeft const):
(WebCore::Layout::InlineDisplayContentBuilder::setLogicalRight const):
(WebCore::Layout::marginLeftInInlineDirection): Deleted.
(WebCore::Layout::marginRightInInlineDirection): Deleted.
(WebCore::Layout::borderLeftInInlineDirection): Deleted.
(WebCore::Layout::borderRightInInlineDirection): Deleted.
(WebCore::Layout::paddingLeftInInlineDirection): Deleted.
(WebCore::Layout::paddingRightInInlineDirection): Deleted.
(WebCore::Layout::InlineDisplayContentBuilder::setLeftForWritingMode const): Deleted.
(WebCore::Layout::InlineDisplayContentBuilder::setRightForWritingMode const): Deleted.
* Source/WebCore/layout/formattingContexts/inline/display/InlineDisplayContentBuilder.h:
* Source/WebCore/layout/formattingContexts/inline/display/InlineDisplayLineBuilder.cpp:
(WebCore::Layout::InlineDisplayLineBuilder::build const):

Canonical link: https://commits.webkit.org/288380@main
